account_invoice_import_invoice2data/models/invoice2data_product_mapping.py

```diff
@@ -0,0 +1,29 @@
+# coding: utf-8
+# Copyright (C) 2021 Essent <http://www.essent.be>
+# @author Stefan Rijnhart <stefan.rijnhart@dynapps.be>
+# @author Robin Conjour <r.conjour@essent.be>
+# @author Pieter Paulussen <pieter.paulussen@dynapps.be>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from openerp import api, fields, models
+
+
+class Invoice2dataProductMapping(models.Model):
+    _name = 'invoice2data.product.mapping'
+    _description = 'Productmapping for invoice2data products'
+
+    account_analytic_id = fields.Many2one(
+        'account.analytic.account', 'Analytic Account')
+    account_id = fields.Many2one(
+        'account.account', 'Account')
+    product_id = fields.Many2one('product.product')
+    rec_name = fields.Char()
+    tax_id = fields.Many2one(
+        'account.tax', string='Taxes',
+        domain=[('type_tax_use', 'in', ('all', 'purchase'))])
+    template_id = fields.Many2one('invoice2data.template')
+
+    @api.onchange('product_id')
+    def _onchange_product_id(self):
+        """ Set account_id and tax_id based on the product_id. """
+        self.account_id = self.product_id.property_account_expense
```

Comment:
Can we prevent this duplication of settings? I would *somewhat* prefer that we require a product to be configured and then always use that product's property_account_expense/supplier taxes, but if that is not possible would you consider to *hide* the account_id/tax_id fields whenever a product is selected on the mapping, and then use the actual product settings at mapping time instead of the values from these fields?

BTW product's supplier_taxes_id is Many2many so if you want to keep a tax field on this model it should probably be a Many2many as well.

Reply:
Changed the mapping fields